cbcontainers/communication/gateway/api_gateway.go

```diff
@@ -21,9 +20,34 @@ type ApiGateway struct {
 	client      *resty.Client
 }
 
-func NewApiGateway(account, cluster string, accessToken string, scheme, host string, port int, adapter string) *ApiGateway {
+func createClient(tlsInsecureSkipVerify bool, rootCAsBundle []byte) (*resty.Client, error) {
 	client := resty.New()
-	client.SetTLSClientConfig(&tls.Config{InsecureSkipVerify: true})
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: tlsInsecureSkipVerify,
+	}
+
+	if len(rootCAsBundle) > 0 {
+		rootCAs := x509.NewCertPool()
+		ok := rootCAs.AppendCertsFromPEM(rootCAsBundle)
+		if !ok {
+			return nil, fmt.Errorf("failed to parse root certificates")
+		}
+
+		tlsConfig.RootCAs = rootCAs
+	}
+
+	client.SetTLSClientConfig(&tlsConfig)
+	return client, nil
+}
+
+func NewApiGateway(account, cluster string, accessToken string, scheme, host string, port int, adapter string,
+	tlsInsecureSkipVerify bool, rootCAsBundle []byte) (*ApiGateway, error) {
```

Comment:
this method has quite a lot of parameters already. might be a good idea to extract them into a struct. something like:
```go
func NewApiGateway(cfg *GatewayConfig) (*ApiGateway, error)
```

might not be now, I interact with this in #52 , so I can do it will my changes. WDYT?

Reply:
nvm this comment now, I just saw the `Builder`. I think it's neat.